pxtcompiler/emitter/driver.ts

```diff
@@ -333,6 +340,14 @@ namespace ts.pxtc {
             tsFiles.push("main.ts")
             hasMain = true;
         }
+
+        // run post-processing code last, if present
+        const post_idx = tsFiles.indexOf("_onCodeStop.ts")
+        if (post_idx >= 0) {
+            tsFiles.splice(post_idx, 1)
+            tsFiles.push("_onCodeStop.ts")
```

Comment:
what happens when you have events?

Reply:
in the pxt-minecraft side, we only send the "onCodeStop" message if no events are registered. basically the minecraft team wants: 1. always send an event before onStart and 2. send "onCodeEnd" after the onStart code, only if no events are registered. i think codeStart might be useful for us in other scenarios but codeStop feels minecraft-HOC-specific to me, since there's no real concept of "code stop" if events are registered

i could rename these to beforeMain/afterMain or something to be clearer? or move codeStop somewhere else maybe...